workflows/differentialabundance.nf

```diff
@@ -60,34 +63,54 @@ include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoft
 
 // Info required for completion email and summary
 def multiqc_report = []
-
+//TODO: make lowercase? https://nf-co.re/docs/contributing/guidelines/requirements/workflow_name
 workflow DIFFERENTIALABUNDANCE {
 
     ch_versions = Channel.empty()
 
     //
     // SUBWORKFLOW: Read in samplesheet, validate and stage input files
     //
-    INPUT_CHECK (
-        ch_input
-    )
-    ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
+    //TODO
+    //INPUT_CHECK (
+    //    ch_input
+    //)
+    //ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
 
     //
-    // MODULE: Run FastQC
+    // MODULE: Run ImportMergedCounts
     //
-    FASTQC (
-        INPUT_CHECK.out.reads
+    IMPORTMERGEDCOUNTS (
+        ch_samplesheet,
+        ch_merged_gene_counts
     )
-    ch_versions = ch_versions.mix(FASTQC.out.versions.first())
-
-    CUSTOM_DUMPSOFTWAREVERSIONS (
-        ch_versions.unique().collectFile(name: 'collated_versions.yml')
+    ch_modified_samplesheet = IMPORTMERGEDCOUNTS.out.ch_modified_samplesheet //file(IMPORTMERGEDCOUNTS.out.ch_modified_samplesheet.first())
+    ch_modified_gene_counts = IMPORTMERGEDCOUNTS.out.ch_modified_gene_counts //file(IMPORTMERGEDCOUNTS.out.ch_modified_gene_counts.first())
+    ch_versions = ch_versions.mix(IMPORTMERGEDCOUNTS.out.versions)
+
+    ch_input = Channel.fromPath(params.contrast_meta)
+        .splitCsv ( header:true, sep:',' )
+        .map{
+            tuple(
+                it,
+                ch_modified_samplesheet.val,
+                ch_modified_gene_counts.val
+                )
+        }
```

Comment:
```suggestion
        .map{
            meta_mapping (it)
        }
```

Reply:
Here it's easier to define a function for mapping to the meta object that you need for the contrast. You can see examples of that here: https://github.com/nf-core/airrflow/blob/8c1185f7d659d7797a66b541bb27aa5861d34991/subworkflows/local/input_check.nf#L24